src/workflows/annotation/celltypist/main.nf

```diff
@@ -0,0 +1,125 @@
+workflow run_wf {
+  take:
+    input_ch
+
+  main:
+    
+    output_ch = input_ch
+        // Set aside the output for this workflow to avoid conflicts
+        | map {id, state -> 
+          def new_state = state + ["workflow_output": state.output]
+          [id, new_state]
+        }
+        // Log normalize query dataset to target sum of 10000
+        | normalize_total.run(
+          fromState: { id, state -> [
+            "input": state.input,
+            "modality": state.modality,
+            "input_layer": state.input_layer,
+          ]},
+          args: [
+            "output_layer": "normalized_10k",
+            "target_sum": "10000",
+          ],
+          toState: [
+            "input": "output",
+          ]
+        )
+
+        | log1p.run( 
+          fromState: { id, state -> [
+            "input": state.input,
+            "modality": state.modality
+          ]},
+          args: [
+            "input_layer": "normalized_10k",
+            "output_layer": "log_normalized_10k",
+          ],
+          toState: [
+            "input": "output"
+          ]
+        )
+        | delete_layer.run(
+          fromState: { id, state -> [
+            "input": state.input,
+            "modality": state.modality
+          ]},
+          args: [
+            "layer": "normalized_10k"
+          ],
+          toState: [
+            "input": "output"
+          ]
+        )
+        // Log normalize reference dataset to target sum of 10000
```

Comment:
Can you run the two log normalizations in parallel?
Something like this should work:

```
log_query_ch = input_ch
   | normalize_total.run(...)
   | log1p.run(...)
   | delete_layer.run(...)

log_ref_ch = input_ch
   | normalize_total.run(...)
   | log1p.run(...)
   | delete_layer.run(...)
 
output_ch = log_query_ch.join(log_ref_ch, failOnMismatch: true, failOnDuplicate: true)
  | map {id, query_state, ref_state ->
     def newState = [query_state + ["reference": ref_state.output]]
     return [id, newState]
  }
```